LearningMissionSimulation/AttributeGenerator.cs

```diff
@@ -300,17 +302,21 @@ public static string GetCoverLetter()
         }
 
         // Returns a random building number
-        public static int GetBuildingNumber()
+        public static int GetBuildingNumber(byte minBuildingNumber, byte maxBuildingNumber)
         {
-            //not implemented yet!
-            return 0;
+            minBuildingNumber = Math.Max(SimulationConstants.MinBuildingNumber, minBuildingNumber);
+            maxBuildingNumber = Math.Min(SimulationConstants.MaxBuildingNumber, maxBuildingNumber);
+
+            return (byte)random.Next(minBuildingNumber, maxBuildingNumber + 1);
         }
 
-        // Returns a random apartment number
-        public static int GetApartmentNumber()
+        // Returns a Random apartment number
+        public static int GetApartmentNumber(byte minApartmentNumber, byte maxApartmentNumber)
```

Comment:
We don't need that flexibility, remove parameters. Just use the constants MinApartmentNumber, MaxApartmentNumber that you defined in the SimulationConstants file: random.Next(MinApartmentNumber, MaxApartmentNumber). Use the same logic in GetBuildingNumber.